scripts/test.sh

```diff
@@ -390,9 +391,13 @@ function cov_pass {
 ######### Code formatting checkers #############################################
 
 function shellcheck_pass {
-  if tool_exists "shellcheck" "https://github.com/koalaman/shellcheck#installing"; then
-    generic_checker run shellcheck -fgcc scripts/*.sh
+  if ! tool_exists "shellcheck" "https://github.com/koalaman/shellcheck#installing"; then
+    wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${SHELLCHECK_VERSION}/shellcheck-${SHELLCHECK_VERSION}.linux.x86_64.tar.xz" | tar -xJv
+    mkdir -p ${HOME}/.local/bin
```

Comment:
I think it's a bad pattern when some innocent test script is playing with operation-system wide or user-wide settings / layout and can substitute scripts on the PATH with vulnerable code. 

As a risk we could imagine here another script that expect different version of shellcheck and them racing about the content of ${HOME}/.local/bin. 

I think everything that ./test.sh script is doing should be limited to the: 
  - `./etcd` directory
  - unique `/tmp` subdirectory that should be cleaned afterwards
  - golang-level module caches that golang takes responsibility for versioning and race-protection. 

Although this code would work in github workflows, I think it's not recommended for developer's environment. 

I would go in the direction of @serathius proposal. 
Maybe even `./etcd/.tmp/bin/shellcheck` and making sure .tmp is on the `.gitignore`.